pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -216,9 +216,12 @@ func (gce *GCECloud) ensureExternalLoadBalancer(clusterName, clusterID string, a
 			if supportsNodesHealthCheck {
 				hcToDelete = makeHttpHealthCheck(MakeNodesHealthCheckName(clusterID), GetNodesHealthCheckPath(), GetNodesHealthCheckPort())
 			}
-			tpNeedsRecreation = true
+
+			// [just reference, will be deleted] tpNeedsRecreation = true
+		}
+		if tpNeedsRecreation || hcLocalTrafficExisting == nil {
```

Comment:
What is the reasoning behind this? A comment here would be very helpful.

Reply:
Comment added. Not sure if I made it clear. Let me know if you have a better way to interpret it.